commit 0ce7556e1b4a109369be11876260b3649e8b0772
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Fri Mar 9 12:01:00 2018 +0000

    lan78xx: Read initial EEE status from DT
    
    Add two new DT properties:
    * microchip,eee-enabled  - a boolean to enable EEE
    * microchip,tx-lpi-timer - time in microseconds to wait before entering
                               low power state
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index 85def9e960ca..60955d04f75b 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -2538,6 +2538,22 @@ static int lan78xx_open(struct net_device *net)
 	if (ret < 0)
 		goto done;
 
+	if (of_property_read_bool(dev->udev->dev.of_node,
+				  "microchip,eee-enabled")) {
+		struct ethtool_eee edata;
+		memset(&edata, 0, sizeof(edata));
+		edata.cmd = ETHTOOL_SEEE;
+		edata.advertised = ADVERTISED_1000baseT_Full |
+				   ADVERTISED_100baseT_Full;
+		edata.eee_enabled = true;
+		edata.tx_lpi_enabled = true;
+		if (of_property_read_u32(dev->udev->dev.of_node,
+					 "microchip,tx-lpi-timer",
+					 &edata.tx_lpi_timer))
+			edata.tx_lpi_timer = 600; /* non-aggressive */
+		(void)lan78xx_set_eee(net, &edata);
+	}
+
 	/* for Link Check */
 	if (dev->urb_intr) {
 		ret = usb_submit_urb(dev->urb_intr, GFP_KERNEL);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1168_0ce7556e1b4a109369be11876260b3649e8b0772.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1168_0ce7556e1b4a109369be11876260b3649e8b0772.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
